### PR TITLE
Add E2E tests logging mechanism

### PR DESCRIPTION
**Pull request by jakalkhalili**
Due to the lack of the E2E tests information about what's gone wrong when a test fails, we decided to improve the logging mechanism:
* After a test fails, the pods in the namespace will be listed and printed
* After a test fails, the Kubernetes events in the namespace will be listed and printed
* After a test fails, the operator logs will be printed

### CODE REVIEW

File: test/e2e/operator.go [hunk 1]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
 
-func failTestAndPrintLogs(t *testing.T, namespace string, err error) {
-	operatorPod := getOperatorPod(t, namespace)
-	logs, logsErr := getOperatorLogs(*operatorPod)
-	if logsErr != nil {
-		t.Errorf("Couldn't get pod logs: %s", logsErr)
+func printOperatorLogs(t *testing.T, namespace string) {
+	logs, err := getOperatorLogs(namespace)
+	if err != nil {
+		t.Errorf("Couldn't get pod logs: %s", err)
```
Comment (tomaszsek) [inline test/e2e/operator.go]: `Couldn't get the operator pod logs`

File: test/e2e/operator.go [hunk 2]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
+	}
+}
+
+func getKubernetesPods(namespace string) ([]v1.Pod, error) {
+	pods, err := framework.Global.KubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return pods.Items, nil
```
Comment (tomaszsek) [inline test/e2e/operator.go]: could be simpler:
`framework.Global.KubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{})`

File: test/e2e/operator.go [hunk 3]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
+}
+
+func printKubernetesPods(t *testing.T, namespace string) {
+	podList, err := getKubernetesPods(namespace)
+	if err != nil {
+		t.Errorf("Couldn't get kubernetes pods: %s", err)
+	}
+
```
Comment (tomaszsek) [inline test/e2e/operator.go]: add `t.Logf("All pods in the namespace:\n")`

File: test/e2e/operator.go [hunk 4]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
+	for _, pod := range podList {
+		t.Logf("%+v\n\n", pod)
+	}
+}
+
+func cleanupAndShowLogs(t *testing.T, ctx *framework.TestCtx) {
```
Comment (tomaszsek) [inline test/e2e/operator.go]: `cleanupAndShowLogs` -> `showLogsAndCleanup`

File: test/e2e/operator.go [hunk 5]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
+	if t.Failed() {
+		t.Error("Test failed. Bellow here you can check logs:")
```
Comment (tomaszsek) [inline test/e2e/operator.go]: `t.Error` -> `t.Log`

File: test/e2e/operator.go [hunk 6]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
+		t.Fail()
```
Comment (tomaszsek) [inline test/e2e/operator.go]: remove because the test has already failed

File: test/e2e/operator.go [hunk 7]
```diff
@@ -59,14 +68,83 @@ func getOperatorLogs(pod v1.Pod) (string, error) {
...
+		}
+
+		t.Logf("Operator logs in '%s' namespace:\n", namespace)
+		printOperatorLogs(t, namespace)
+
+		t.Logf("Kubernetes events in '%s' namespace:\n", namespace)
+		printKubernetesEvents(t, namespace)
+
+		t.Logf("Pods in '%s' namespace:\n", namespace)
+		printKubernetesPods(t, namespace)
```
Comment (tomaszsek) [inline test/e2e/operator.go]: These logs should be inside these functions.

File: test/e2e/operator.go (hunk 4 above)
Comment (jakalkhalili) [reply test/e2e/operator.go]: Its already on line 140

**Pull request merged**